jupyterhub_magpie_authenticator/jupyterhub_magpie_authenticator.py

```diff
@@ -73,4 +80,20 @@ async def authenticate(self, handler, data):
                                    expires=cookie.expires,
                                    path=cookie.path,
                                    secure=cookie.secure)
+            if self.enable_auth_state:
+                return {"name": data['username'], "auth_state": {"magpie_cookies": response.cookies.get_dict()}}
```

Comment:
Maybe add an `else` here so it's clearer it's either one form of `return` or the other.  Else I usually prefer to have only one `return` and the end to avoid multiple exit path in a function, which sometime makes the code more complex to understand (not in this case, I mean in general):

```python
if self.enable_auth_state:
    result = {"name": data['username'], "auth_state": {"magpie_cookies": response.cookies.get_dict()}}
else
    result = data['username']

return result
```

This brings to another remark, the return type is either a dict or just a string.  Should this be harmonized to always return a dict?  Ignore this comment if it breaks the "regular existing" workflow the code.

Reply:
> the return type is either a dict or just a string. Should this be harmonized to always return a dict?

The `authenticate` function that this overrides returns a `str`, `dict` or `None`:

https://github.com/jupyterhub/jupyterhub/blob/main/jupyterhub/auth.py#L621-L622

Each return type is handled differently by the calling function.

For our case, I don't think that it would be a problem to change it to always return a `dict` or `None` but I don't think it's necessary